New registration page

Flaburgan opened the pull request:
Following the discussions about [the import feature](https://discourse.diasporafoundation.org/t/backup-and-restore-account-migration/544/166?u=flaburgan), here is a new registration pages, revamped, responsive, and stating to the users that they will be able to import once registered.

![Capture d’écran 2021-08-29 à 23 07 30](https://user-images.githubusercontent.com/930064/131265358-203f7752-fc4c-4750-955d-97ed1686dfcc.png)

Do you like it better with the ball?

![Capture d’écran 2021-08-29 à 23 12 31](https://user-images.githubusercontent.com/930064/131265486-d04b3c72-b60c-465a-abdc-6aea7ad95810.png)

![image](https://user-images.githubusercontent.com/930064/131265227-d8371253-86c7-407e-be28-f9a7a24e42d2.png)

This PR introduces the basis of a very simple CSS framework. I feel like we don't need bootstrap anymore, it looks old and was mainly used for its grid which is obsolete now with flexboxes.

diaspora-code-review commented on app/assets/stylesheets/framework/_framework.scss:
```diff
@@ -0,0 +1,11 @@
+@import "fonts";
```
Prefer single quoted strings

diaspora-code-review commented on app/assets/stylesheets/framework/_framework.scss:
```diff
@@ -0,0 +1,11 @@
+@import "fonts";
+@import "variables";
```
Prefer single quoted strings

diaspora-code-review commented on app/assets/stylesheets/framework/_framework.scss:
```diff
@@ -0,0 +1,11 @@
+@import "fonts";
+@import "variables";
+@import "ui";
```
Prefer single quoted strings

diaspora-code-review commented on app/assets/stylesheets/framework/_framework.scss:
```diff
@@ -0,0 +1,11 @@
+@import "fonts";
+@import "variables";
+@import "ui";
+@import "layout";
```
Prefer single quoted strings

diaspora-code-review commented on app/assets/stylesheets/framework/_framework.scss:
```diff
@@ -0,0 +1,11 @@
+@import "variables";
+@import "ui";
+@import "layout";
+@import "form";
```
Prefer single quoted strings

diaspora-code-review commented on app/assets/stylesheets/framework/fonts.scss:
```diff
@@ -0,0 +1,20 @@
+  font-family: 'Open Sans';
+  font-style: normal;
+  font-weight: 400;
+  src: local("Open Sans Regular"), local("OpenSans-Regular"), url(/assets/fonts/open-sans-400.woff2) format("woff2");
```
Prefer single quoted strings

diaspora-code-review commented on app/assets/stylesheets/framework/fonts.scss:
```diff
@@ -0,0 +1,20 @@
+  font-family: 'Open Sans';
+  font-style: italic;
+  font-weight: 400;
+  src: local("Open Sans Italic"), local("OpenSans-Italic"), url(/assets/fonts/open-sans-400italic.woff2) format("woff2");
```
Prefer single quoted strings

diaspora-code-review commented on app/assets/stylesheets/framework/fonts.scss:
```diff
@@ -0,0 +1,20 @@
+  font-family: 'Open Sans';
+  font-style: normal;
+  font-weight: 600;
+  src: local("Open Sans SemiBold"), local("OpenSans-SemiBold"), url(/assets/fonts/open-sans-600.woff2) format("woff2");
```
Prefer single quoted strings

diaspora-code-review commented on app/assets/stylesheets/framework/form.scss:
```diff
@@ -0,0 +1,44 @@
+[type=email],
+[type=password] {
+  width: 320px;
+  box-shadow: inset 1px 0 1px #ddd;
```
Color literals like `#ddd` should only be used in variable declarations; they should be referred to via variable everywhere else.

diaspora-code-review commented on app/assets/stylesheets/framework/form.scss:
```diff
@@ -0,0 +1,44 @@
+}
+
+@media screen and (max-width: 700px) {
+	[type=text],
```
Line should be indented with spaces, not tabs

diaspora-code-review commented on app/assets/stylesheets/framework/form.scss:
```diff
@@ -0,0 +1,44 @@
+select,
+textarea {
+  // Fix dirty box model rendering on inputs
+  -moz-box-sizing: border-box;
```
- Properties should be ordered background-color, border, border-radius, -moz-box-sizing, -webkit-box-sizing, box-sizing, font-size, padding
- Avoid vendor prefixes.

diaspora-code-review commented on app/assets/stylesheets/framework/form.scss:
```diff
@@ -0,0 +1,44 @@
+[type=text],
+[type=email],
+[type=password] {
+  width: 320px;
```
Properties should be ordered box-shadow, width

diaspora-code-review commented on app/assets/stylesheets/framework/form.scss:
```diff
@@ -0,0 +1,44 @@
+textarea {
+  // Fix dirty box model rendering on inputs
+  -moz-box-sizing: border-box;
+  -webkit-box-sizing: border-box;
```
Avoid vendor prefixes.

diaspora-code-review commented on app/assets/stylesheets/framework/layout.scss:
```diff
@@ -0,0 +1,20 @@
+// This cleans the famous children "margin-top" problem
+.small-container::before,
+.small-container::after {
+  display: table;
```
Properties should be ordered content, display

diaspora-code-review commented on app/assets/stylesheets/framework/layout.scss:
```diff
@@ -0,0 +1,20 @@
+}
+
+.small-container {
+  max-width: 800px;
```
Properties should be ordered margin, max-width

diaspora-code-review commented on app/assets/stylesheets/framework/layout.scss:
```diff
@@ -0,0 +1,20 @@
+.small-container::before,
+.small-container::after {
+  display: table;
+  content: " ";
```
Prefer single quoted strings

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+	&.disabled,
+	&.disabled:hover {
+		background-color: #ccc;
+		color: white;
```
- Color `white` should be written in hexadecimal form as `#ffffff`
- Color literals like `white` should only be used in variable declarations; they should be referred to via variable everywhere else.

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+	&:disabled:hover,
+	&.disabled,
+	&.disabled:hover {
+		background-color: #ccc;
```
Color literals like `#ccc` should only be used in variable declarations; they should be referred to via variable everywhere else.

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+
+.bttn {
+  border: 0;
+	border-radius: $radius;
```
Line should be indented with spaces, not tabs

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+.bttn {
+  border: 0;
+	border-radius: $radius;
+	cursor: pointer;
```
Line should be indented with spaces, not tabs

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+  border: 0;
+	border-radius: $radius;
+	cursor: pointer;
+	display: inline-block;
```
Line should be indented with spaces, not tabs

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+	border-radius: $radius;
+	cursor: pointer;
+	display: inline-block;
+	font-weight: bold;
```
Line should be indented with spaces, not tabs

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+	cursor: pointer;
+	display: inline-block;
+	font-weight: bold;
+	line-height: 22px;
```
Line should be indented with spaces, not tabs

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+	font-weight: bold;
+	line-height: 22px;
+  padding: 8px 20px;
+	text-decoration: none;
```
Line should be indented with spaces, not tabs

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+  padding: 8px 20px;
+	text-decoration: none;
+
+	&:disabled,
```
Line should be indented with spaces, not tabs

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+		cursor: wait;
+	}
+
+	&:focus {
```
Line should be indented with spaces, not tabs

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+}
+
+.bttn-primary {
+	@extend .bttn;
```
- Line should be indented with spaces, not tabs
- Prefer using placeholder selectors (e.g. %some-placeholder) with @extend

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+.bttn-primary {
+	@extend .bttn;
+
+	&,
```
Line should be indented with spaces, not tabs

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+		color: $primary-text;
+	}
+
+	&:hover,
```
Line should be indented with spaces, not tabs

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+input[type=submit],
+input[type=button],
+input[type=file] {
+  font-family: inherit;
```
Properties should be ordered -webkit-appearance, font-family

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+
+.advice {
+  color: $subtle;
+  font-style: italic;
```
Properties should be ordered color, font-size, font-style

diaspora-code-review commented on app/assets/stylesheets/framework/ui.scss:
```diff
@@ -0,0 +1,79 @@
+  margin-bottom: $double-margin;
+}
+
+button,
```
Avoid qualifying attribute selectors with an element.

diaspora-code-review commented on app/assets/stylesheets/registration.scss:
```diff
@@ -1,54 +1,48 @@
-    .v-center {
-      height: auto;
+    @media screen and (max-width: $breakpoint) {
+      margin: 2rem 0;
```
Properties should be ordered flex-direction, margin

diaspora-code-review commented on app/assets/stylesheets/registration.scss:
```diff
@@ -1,54 +1,48 @@
-    line-height: $line-height-base;
-    padding-left: 130px;
+    .captcha-img {
+      order: 2;
```
Properties should be ordered margin-left, order

diaspora-code-review commented on app/assets/stylesheets/registration.scss:
```diff
@@ -1,54 +1,48 @@
+    width: 240px;
+
+    @media screen and (max-width: $breakpoint) {
+      width: 100%;
```
Properties should be ordered margin-bottom, margin-left, width

diaspora-code-review commented on app/assets/stylesheets/framework/layout.scss:
```diff
@@ -0,0 +1,20 @@
+// This cleans the famous children "margin-top" problem
+.small-container::before,
+.small-container::after {
+	content: " ";
```
- Line should be indented with spaces, not tabs
- Prefer single quoted strings

tclaus commented:
$: pronto run --unstaged -c upstream/develop

One of my favorites..

SuperTux88 commented on .scss-lint.yml:
```diff
@@ -188,7 +188,7 @@ linters:
   StringQuotes:
     enabled: true
-    style: single_quotes # or double_quotes
+    style: double_quotes
```
If pronto complains a lot it isn't a solution to just change the config. If you think some of the settings don't make sense and we should change this, we can discuss this, but not as a drive-by of this PR.

Especially this one affects all existing files which previously were created with the old rule.

SuperTux88 commented on app/assets/stylesheets/framework/fonts.scss:
```diff
@@ -0,0 +1,20 @@
+@font-face {
```
I don't quite get why you need new fonts only for the registration page. I would think that the font we use on the rest of the pages also would work on a new registration page. If we want to change the font, we should change it everywhere, but again, this doesn't feel right as a drive-by with a new registration-page.

Flaburgan replied:
Yeah as I said I want to rework the design but I thought about going page per page, but maybe it shouldn't be part of that PR.

SuperTux88 commented on app/assets/stylesheets/framework/variables.scss:
```diff
@@ -0,0 +1,27 @@
+$breakpoint: 700px;
+
+$text: #444;
```
This creates completely new variables which ignore all existing color and also doesn't work with the other themes.

goobertron replied:
I wouldn't think the sign-up page would need to work with themes, because a not-yet-registered person won't have set a theme. But it certainly makes sense to use existing variables wherever possible, which will make changing the UI across the app easier in future.

SuperTux88 replied:
A podmin can set a default theme in the diaspora.toml (which I have done on my pod), and this should then still work for the registration page.

goobertron replied:
Ah, thanks. I didn't realise that.

SuperTux88 reviewed (commented):
First: I like the screenshots, and I also like it with the ball (since there is space).

But I don't like the color. We don't really use this color anywhere else (except for this one place which you see once and never again), and the orange looks really strange and also looks a bit like a warning (first I thought that all fields are orange because they are empty). I would use the primary color (of the existing themes, which on the default theme is blue, on the colored themes it's the corresponding color). That way it doesn't look as warnings anymore.

The code comments aren't a full review yet, it's only what I noticed when having a quick look on it. I think if you just want to have a new registration page (which I don't have anything against it), it should work with the existing styles/themes. But the code looks more like you started to create a completely new style/theme/frontend, which I think shouldn't be done as a drive-by with a new registration page. There was a plan to create a new frontend, but I think this needs more discussion first, for example here: https://discourse.diasporafoundation.org/t/choosing-new-frontend-framework/1070 But I think that's a completely different topic/discussion.

Flaburgan commented:
>But I don't like the color. We don't really use this color anywhere else (except for this one place which you see once and never again), and the orange looks really strange and also looks a bit like a warning (first I thought that all fields are orange because they are empty). I would use the primary color (of the existing themes, which on the default theme is blue, on the colored themes it's the corresponding color). That way it doesn't look as warnings anymore.

Yeah totally agree on this, I picked that color a bit randomly, we can stay with the primary blue at the moment, that is fine.

> I think if you just want to have a new registration page (which I don't have anything against it), it should work with the existing styles/themes. But the code looks more like you started to create a completely new style/theme/frontend, which I think shouldn't be done as a drive-by with a new registration page. There was a plan to create a new frontend, but I think this needs more discussion first, for example here: https://discourse.diasporafoundation.org/t/choosing-new-frontend-framework/1070 But I think that's a completely different topic/discussion.

Indeed, my idea here is to introduce a new design with an homemade minimalist CSS framework. Bootstrap is heavy, old, outdated and not needed anymore (we have flexboxes now, no need for a grid). Please note that this isn't about a new JS framework though, so IMO it is unrelated to the discourse discussion you linked.

I went with this idea of slowly introducing that new framework page per page instead of trying to rewrite everything, but maybe now (and here?) isn't the time to do that?

goobertron commented:
Thanks for doing this. I'm happy with the design, and it's good that it's responsive. I also question the orange, which I think is only used elsewhere for the 'getting started' banner. And if we're going to change the UI, I think it makes sense to do it in a single release for the whole UI, so I'd prefer not to introduce new typefaces just for this page. It should blend with the rest of the UI.

I only have the screen-shots to refer to, not being able to test this. Do the vertical lines to the left of the input fields disappear when valid data are entered into the fields? If not, what are they for? Their presences suggests some function, so they could be confusing if they're mere decoration.

Let me know when the text has been moved into a locale file, and I'll edit it/polish the English then.

Flaburgan commented:
Okay so no more orange, and no new font: 
![image](https://user-images.githubusercontent.com/930064/133826752-42f2f984-7ecd-4226-81d0-ff6e7ff78081.png)

Flaburgan commented:
How do you want to proceed with this idea of a custom minimalist CSS framework then? Can I start it in that PR?

SuperTux88 commented:
> Do the vertical lines to the left of the input fields disappear when valid data are entered into the fields? If not, what are they for? Their presences suggests some function, so they could be confusing if they're mere decoration.

I thought the same first, because the orange looked like a warning/error because it's not valid yet. But they are purely decoration, which I don't mind, as long as it's not in a warning-color (well, if a podmin creates a orange theme they might as well be orange if the whole page is orange, but not if it's the only color on the page).

> Okay so no more orange, and no new font

:+1: 

> How do you want to proceed with this idea of a custom minimalist CSS framework then? Can I start it in that PR?

We have bootstrap already anyway, so there is no harm of using it for all pages. As said, rewriting the complete frontend is a bigger task, and probably should have a bit more discussions than just as a drive-by of a new registration page.

Also as goob said "And if we're going to change the UI, I think it makes sense to do it in a single release for the whole UI, so I'd prefer not to introduce new typefaces just for this page. It should blend with the rest of the UI." it should at least be somewhat consistent, so redoing the UI is probably more something that we should start working on for 0.9. It doesn't need to be completely done for ALL the pages, and we can do it page by page on the develop branch for 0.9 once we start working on the new UI. But at least most pages should then be done before the release, so not releasing a half done new UI, and for 0.8 I think we should focus on the migration and then release and not start another big topic.

Also even if you only talk about the styles, it's probably still affected by how we are doing the new frontend, if a new frontend is done using a separate project and then only using the API, it's probably also changing how it's built. And I for example also would like to use CSS-variables instead of SCSS so color themes can be done without pre-compiling the whole styles x times.

So all that needs a bit more thinking and planning and not just start doing something as drive-by.

goobertron commented:
> How do you want to proceed with this idea of a custom minimalist CSS framework then? 

I think a Discourse discussion would be the best place to start.

Flaburgan commented:
>We have bootstrap already anyway, so there is no harm of using it for all pages. As said, rewriting the complete frontend is a bigger task, and probably should have a bit more discussions than just as a drive-by of a new registration page.

Hm okay I will wait then. So I will also revert the button and the form fields still. Can I still use flexboxes instead of the bootstrap grid please? This is so much easier.

SuperTux88 commented:
> Can I still use flexboxes instead of the bootstrap grid please? This is so much easier.

Sure, why not :+1:

Flaburgan committed:
New registration page, mentionning the import, fixes #8009

Flaburgan commented:
New code using Bootstrap for fields and buttons:
![image](https://user-images.githubusercontent.com/930064/133969300-0877c326-7819-4480-ab4d-8cd240ff1e1b.png)

I kept the larger font, it really was too small otherwise, and this is a complain I'm hearing a lot about diaspora* already (stuck in 13px while minimum screens are 1920px now).

Flaburgan commented:
Okay, I kept the bare minimum CSS to have a design OK, new translations are added and tests are green (without having to touch them \o/). So this looks ready to review.

goobertron commented on config/locales/diaspora/en.yml:
```diff
@@ -1091,18 +1091,23 @@ en:
 
   registrations:
     new:
+      title: "Join the community!"
```
I think it's worth specifying: "Join the diaspora* community!"

Flaburgan replied:
OK.

goobertron commented on config/locales/diaspora/en.yml:
```diff
@@ -1091,18 +1091,23 @@ en:
   registrations:
     new:
+      title: "Join the community!"
+      subtitle: "You're about to create an account on %{podname}. One diaspora* account is enough to reach the whole network, you don't need to create a new account here if you already have one on another pod."
```
A couple of minor alterations: "You're about to create an account on %{podname}. A single account is enough to reach the whole diaspora* network. You don't need to create a new account here if you already have one on another pod."

Flaburgan replied:
Thanks, that's better.

goobertron commented on config/locales/diaspora/en.yml:
```diff
@@ -1091,18 +1091,23 @@ en:
       enter_password_again: "Enter the same password as before"
       sign_up: "Create account"
       email: "Email"
+      email_advice: "Your e-mail will never be shared with anyone and will only be used to allow you to recover your account or send notifications."
```
I suggest reordering as follows: "Your e-mail will never be shared with anyone and will only be used to send notifications and to enable you to recover your account."

goobertron replied:
> notifications could be seen as spam

Notifications are entirely optional, so not spam. If you are concerned about that, you could expand this to 'to send the notifications you request'. I think notifications should go first as that's the normal/most common use for the email address, whereas recovering the account is a (hopefully) rare thing. At the very least, please change 'or send notifications' at the end to 'and to send notifications'.

Flaburgan replied:
Yeah I changed it to "Your e-mail will never be shared with anyone and will only be used to enable you to recover your account and to send notifications."

goobertron commented on config/locales/diaspora/en.yml:
```diff
@@ -1091,18 +1091,23 @@ en:
       email: "Email"
+      email_advice: "Your e-mail will never be shared with anyone and will only be used to allow you to recover your account or send notifications."
       username: "Username"
+      username_advice: "Your username is unique and can't be changed. It allows others to find you in the network."
```
"Your username is unique and allows others to find you in the network. It can't be changed once set."

Flaburgan replied:
Better.

goobertron commented on config/locales/diaspora/en.yml:
```diff
@@ -1091,18 +1091,23 @@ en:
       submitting: "Submitting..."
       terms: "By creating an account you accept the %{terms_link}."
       terms_link: "terms of service"
+      import_indication: "You already have a diaspora* account and you want to migrate it on %{podname}? Please sign up first and then import your previous account from the settings."
```
It's difficult to reword this and keep it simple. This is my best effort so far: "If you want to migrate  an existing diaspora* account to  %{podname}, please sign up first and then go to the user settings page to import your previous account."

Flaburgan replied:
Keeping the question makes it more dynamic imo. "You already have a diaspora* account and you want to migrate it on %{podname}? Please sign up first and then go to the user settings page to import your previous account."
Is that okay?

goobertron replied:
""You already have a diaspora* account and you want to migrate it on %{podname}?" Just doesn't read well in English, I'm afraid. I can't think of a way of phrasing it well. Let me think again...

"Do you want to migrate an existing diaspora* account to %{podname}? If so, sign up first and then go to the user settings page to import your previous account."

That works. Does it seem dynamic enough to you?

goobertron reviewed (commented):
With apologies for taking a long time, here are my suggestions for rewording the text.

Flaburgan commented:
Here is the last rendering with goob's changes:
![image](https://user-images.githubusercontent.com/930064/136098801-cc4cd2c6-427c-4736-a1e5-495691be7418.png)

Flaburgan commented:
This is rebased now.

Flaburgan commented:
I would like this to be merged because I think it really improves the registration process, explaining what is a diaspora* id etc.
If we're not sure about the migration, maybe I could strip the panel talking about it, so we can merge this MR without the migration part, and add it in a separate PR? @SuperTux88

Flaburgan commented:
I removed the migration indication and rebased, this could be integrated to the next release.

Flaburgan commented:
I fixed the tests. `minlength` is not fully supported for input type="password" I had to put back the `pattern` attribute as it was done before. I also added a link to the tutorials.

Here is how it looks now:

![image](https://github.com/diaspora/diaspora/assets/930064/90441f15-6c68-4732-9c4a-771d2c1f3cbc)

SuperTux88 reviewed (changes requested):
The page itself looks good, but I'm still not really happy with the CSS changes.

If you want to create a new CSS framework that can also be used by other pages, then the generic variables and selectors (like `body` and `h1`) shouldn't be defined in a file called `registration.scss`.

Then the new CSS framework should be built in a way that it only affects new pages, and doesn't change font sizes and other stuff elsewhere.

And it should still work with color themes, so best use existing color variables, or if new variables are required, at least define them in a way that they can be overwritten by themes.

I'm still not really sure if all of this should be done in this PR, or be moved to a different PR. I have nothing against using flexboxes here (instead of bootstrap grid), since that's a local thing to only this page. But stuff like introducing new default styles is probably better done with a few pages directly (not necessarily all of them at once), to see what should be shared and what not. I like the idea of having a more lightweight CSS framework than using bootstrap, but I think this PR would be easier with using existing styles.

Flaburgan commented on app/assets/stylesheets/registration.scss:
```diff
@@ -1,54 +1,154 @@
+
+$subtle: #777;
+
+$default-size: 16px;
```
Fixed.

Flaburgan commented on app/assets/stylesheets/registration.scss:
```diff
@@ -1,54 +1,154 @@
-    max-width: 100%;
+$breakpoint: 700px;
+
+$subtle: #777;
```
Fixed.

Flaburgan commented on config/locales/diaspora/en.yml:
```diff
@@ -1091,18 +1091,20 @@ en:
       email: "Email"
+      email_advice: "Your e-mail will never be shared with anyone and will only be used to enable you to recover your account and to send notifications."
       username: "Username"
+      username_advice: "Your username is unique and allows others to find you in the network. It can't be changed once set. Only letters, numbers, _ and -."
```
Fixed.

Flaburgan commented on app/assets/stylesheets/registration.scss:
```diff
@@ -1,54 +1,154 @@
+// New default fontsize the page
+html,
+body {
+  font-size: 16px;
```
Fixed.

Flaburgan commented:
I rework my changes, is it better like that?

Flaburgan commented:
@SuperTux88 I rework my changes, is it better like that?

SuperTux88 commented on app/views/registrations/_new.erb:
```diff
@@ -0,0 +1,77 @@
+              <label for="user_username"><%= t("registrations.new.username") %></label>
+            </p>
+            <p>
+              <input type="text" id="user_username" name="user[username]" value="<%= @user.username %>" class="form-control" pattern="[A-Za-z0-9_.\-]+" required />@<%= AppConfig.pod_uri.host %>
```
Not a problem on most production pods, but I just noticed that on my local dev-system, this doesn't include the port, which would also be part of the diaspora-ID. What you want to use instead is `AppConfig.bare_pod_uri`, or maybe even `User.diaspora_id_host` (which also includes the `@`).

SuperTux88 commented on app/assets/stylesheets/registration.scss:
```diff
@@ -1,54 +1,79 @@
+  background: $body-bg;
+
+  #app {
+    margin: -10px;
```
On mobile, there is a very small horizontal scrollbar, and I think it has something to do with this, since this makes the page actually 10px bigger (since it increases the size of `#app`), so what you want to do instead is probably reduce the `padding` of `#main` and leave `#app` as it is.

Flaburgan replied:
Ah nice catch, this is a regression on the last change I made. Fixed, thanks for noticing it.

SuperTux88 commented on app/assets/stylesheets/registration.scss:
```diff
@@ -1,54 +1,79 @@
-      margin: 12px 0;
+    .captcha-img {
+      margin-left: 5px;
+      order: 2;
```
I don't know if this is intended, but this only works in the "desktop" (default) view, but not in "mobile" view. Since both should use the same responsive design, I assume this should look always the same on small screens, independent of in which mode you actually are? So you probably just want to delete `app/views/simple_captcha/_simple_captcha.mobile.haml` so it uses the same template as in "desktop" view?

SuperTux88 reviewed (changes requested):
Thanks, yes this looks already a lot better. :+1:  But I still found a few small things.

By the way, you don't need to respond to each of my comments, it's enough if you write a comment if you are done with everything and I also receive emails when you push new stuff. If you really want to react to the specific comments to mark that you have done them, you can maybe just react with a :+1: to the comment, this doesn't send a separate email for each comment to all the watchers of the repo.

Flaburgan committed:
Fix horizontal scroll and captcha on mobile, host for user id
